virtool/uploads/api.py

```diff
@@ -130,23 +125,18 @@ async def get(self) -> Union[r200[FileResponse], r404]:
             200: Successful operation
             404: Not found
         """
-        pg = self.request.app["pg"]
 
         upload_id = int(self.request.match_info["id"])
 
-        upload = await virtool.uploads.db.get(pg, upload_id)
+        upload = await get_data_from_req(self.request).uploads.get(upload_id)
 
         if not upload:
             raise NotFound()
```

Comment:
Instead of this, follow the `ResourceNotFoundError` pattern used in other data layer pieces.

You should only have one call to the data layer that looks like this:
```py
upload_path = await get_data_from_req(self.request).uploads.get_upload_path(
    upload_id
)
```

Reply:
I made get_upload_path as a utility function since I saw the code was being used twice in the jobs_api as well as the view, whereas get(upload_id) is used to get the actual upload. Should I move get_upload_path to utils?